Source/WebCore/style/AnchorPositionEvaluator.h

```diff
@@ -42,16 +45,61 @@ namespace WebCore {
 
 class Document;
 class Element;
+class LayoutPoint;
 class LayoutRect;
 class LayoutSize;
 class RenderBlock;
 class RenderBox;
 class RenderBoxModelObject;
 class RenderElement;
 class RenderStyle;
+class RenderView;
 
 enum CSSPropertyID : uint16_t;
 
+struct AnchorScrollSnapshot {
+    const RenderBox* m_scroller;
```

Comment:
Please use `WeakPtr<const RenderBox>` or `CheckedPtr<const RenderBox>`. The latter will crash in case the RenderBox dies before the snapshot is deleted so requires correct cleanup code.